### DISCUSSION TITLE
How to get selected value in Tree

### DISCUSSION THREAD

**Discussion by nghia-vo** [zauberzeug/nicegui#400] [Q&A] [answered]
This may be a simple question but I can't find the answer in public examples. How to get the value selected by users in a tree. In the example below, GUI issues a notification but I would like to get the value and do something else with it.
```python
from nicegui import ui

ui.tree([
    {'id': 'numbers', 'children': [{'id': '1'}, {'id': '2'}]},
    {'id': 'letters', 'children': [{'id': 'A'}, {'id': 'B'}]},
], label_key='id', on_select=lambda e: ui.notify(e.value))

ui.run() 
````
How to pass e.value to a variable in the main program?

**Comment by rodja** [comment, marked as answer]
You can use any variable, or call a function to process the selection. A simple example:

```py
history = []

ui.tree([
    {'id': 'numbers', 'children': [{'id': '1'}, {'id': '2'}]},
    {'id': 'letters', 'children': [{'id': 'A'}, {'id': 'B'}]},
], label_key='id', on_select=lambda e: history.append(e.value))

ui.timer(1, lambda: print(history, flush=True) if history else None)
```

**Reply by nghia-vo** [reply]
Many thanks. The last line allows to check the result. I only used bare "print" before but didn't see any output which is the reason for the question.
